File: spring/src/main/java/cardsproject/domain/content/DraftPick.java
package cardsproject.domain.content;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "draft_picks")
public class DraftPick {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Integer pickNumber = 0;
    private Integer packNumber = 0;
    private LocalDateTime pickedAt;

    @Column(name = "participant_id")
    private Long participantId;
    @Column(name = "card_id")
    private Long cardId;

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Integer getPickNumber() { return pickNumber; }
    public void setPickNumber(Integer pickNumber) { this.pickNumber = pickNumber; }
    public Integer getPackNumber() { return packNumber; }
    public void setPackNumber(Integer packNumber) { this.packNumber = packNumber; }
    public LocalDateTime getPickedAt() { return pickedAt; }
    public void setPickedAt(LocalDateTime pickedAt) { this.pickedAt = pickedAt; }
    public Long getParticipantId() { return participantId; }
    public void setParticipantId(Long participantId) { this.participantId = participantId; }
    public Long getCardId() { return cardId; }
    public void setCardId(Long cardId) { this.cardId = cardId; }

    // ── Business operations ──────────────────────────────────────────
    @com.fasterxml.jackson.annotation.JsonIgnore
    public Boolean isFirstPick() {
        throw new UnsupportedOperationException("isFirstPick not implemented");
    }

    // ── Validation rules ─────────────────────────────────────────────
    @jakarta.validation.constraints.AssertTrue(message = "Pick number must be greater than zero")
    @com.fasterxml.jackson.annotation.JsonIgnore
    public boolean isPickNumberPositiveValid() {
        return (getPickNumber() == null || getPickNumber() > 0);
    }
    @jakarta.validation.constraints.AssertTrue(message = "Pack number must be between 1 and 3")
    @com.fasterxml.jackson.annotation.JsonIgnore
    public boolean isPackNumberRangeValid() {
        return (getPackNumber() == null || (getPackNumber() >= 1 && getPackNumber() <= 3));
    }
}

File: symfony/src/Entity/Content/DraftPick.php
<?php

namespace App\Entity\Content;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;
use App\Repository\Content\DraftPickRepository;
use App\Entity\Cards\Card;

#[ORM\Entity(repositoryClass: DraftPickRepository::class)]
#[ORM\Table(name: 'draft_pick')]
class DraftPick
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    #[Groups(['draftPick:read'])]
    private ?int $id = null;

    #[ORM\Column(type: 'integer')]
    #[Groups(['draftPick:read', 'draftPick:write'])]
    private int $pickNumber = 0;

    #[ORM\Column(type: 'integer')]
    #[Groups(['draftPick:read', 'draftPick:write'])]
    private int $packNumber = 0;

    #[ORM\Column(type: 'datetime', nullable: true)]
    #[Groups(['draftPick:read', 'draftPick:write'])]
    private ?\DateTimeInterface $pickedAt = null;

    #[ORM\ManyToOne(targetEntity: DraftParticipant::class, inversedBy: 'picks')]
    #[ORM\JoinColumn(nullable: false)]
    private ?DraftParticipant $participant = null;

    #[ORM\ManyToOne(targetEntity: Card::class, inversedBy: 'draft_picks')]
    #[ORM\JoinColumn(nullable: false)]
    private ?Card $card = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getPickNumber(): int
    {
        return $this->pickNumber;
    }

    public function setPickNumber(int $pickNumber): static
    {
        $this->pickNumber = $pickNumber;
        return $this;
    }

    public function getPackNumber(): int
    {
        return $this->packNumber;
    }

    public function setPackNumber(int $packNumber): static
    {
        $this->packNumber = $packNumber;
        return $this;
    }

    public function getPickedAt(): ?\DateTimeInterface
    {
        return $this->pickedAt;
    }

    public function setPickedAt(?\DateTimeInterface $pickedAt): static
    {
        $this->pickedAt = $pickedAt;
        return $this;
    }

    #[Groups(['draftPick:read'])]
    public function getParticipantId(): ?int
    {
        return $this->participant?->getId();
    }

    public function getParticipant(): ?DraftParticipant
    {
        return $this->participant;
    }

    public function setParticipant(?DraftParticipant $participant): static
    {
        $this->participant = $participant;
        return $this;
    }

    #[Groups(['draftPick:read'])]
    public function getCardId(): ?int
    {
        return $this->card?->getId();
    }

    public function getCard(): ?Card
    {
        return $this->card;
    }

    public function setCard(?Card $card): static
    {
        $this->card = $card;
        return $this;
    }

    // ── Business operations ──────────────────────────────────────────

    public function isFirstPick(): void
    {
        throw new \RuntimeException('is_first_pick not implemented');
    }

}
